ONSdigital/takeon-business-layer
# Feature/spp 222 days since

# Description
Create 'Days Since' Derived Questions (survey 999A).

This is implemented for Form 100 of 999A survey

- Added 2 new questions Q3(3213) and Q4(3214) which are derived values of Q1(start date) and Q2(End date) in the database
- This pull request contains the corresponding BL changes of calculation of each derived question Q3 and Q4 number of days since year 1900/01/01 for the corresponding Q1 and Q2 values

# How to test?

- **https://dc-ui-spp222.es-take-on.aws.onsdigital.uk/**
- Go to Survey '999A' and Form 100
- You can see 2 new derived questions Q3 and Q4
Q3 PoP QvQ Derived Question (Q1 Days since 01/01/1900)
Q4 PoP QvQ Derived Question (Q2 Days since 01/01/1900 plus one)

- Enter some date values in Q1 and Q2 in YYYYMMDD format
- You can see the calculated values in Q3 and Q4 and make sure those number of days for Q3 is days since of Q1 and no of days of Q4 are days since of Q2 + 1

# Unit Tests
You can run java unit tests using: mvn clean test

# Code Coverage
You can verify code coverage using: awk -F"," '{ instructions += $4 + $5; covered += $5 } END { print covered, "/", instructions, " instructions covered"; print 100*covered/instructions, "% covered" }' target/site/jacoco/jacoco.csv

# Check Style errors
You can verify check style errors using : mvn package install

# Links
https://collaborate2.ons.gov.uk/jira/browse/SPP-222
